design/authn_azure/authn_azure_solution_design.md

```diff
@@ -0,0 +1,449 @@
+### Affected Components
+
+- Conjur
+
+- DAP
+
+## Test Plan
+
+### Integration tests
+
+#### Functionality
+
+This suite of tests assumes that the Azure authenticator has been configured correctly
+
+|    | Given                                                                                        | When                                                                | Then                                                                                     | Manual / UT / Integration | Status           |
+|----|----------------------------------------------------------------------------------------------|---------------------------------------------------------------------|------------------------------------------------------------------------------------------|---------------------------|------------------|
+| 1  | *Vanilla flow* - a Conjur identity exists for Azure VM and added to privileged layer         | VM sends authentication request with a valid Azure token            | Return Conjur access token is returned in body and action is logged                      | Integration               | - [ ]            |
+| 2  | A Conjur Host identity for Azure VM, added to privileged layer and has permissions on secret | VM sends request with valid Azure token to fetch secret             | Azure resource can successfully send GET request for a secret and action is logged       | Integration               | - [ ]            |
+| 3  | No Conjur Host identity exists for Azure VM                                                  | VM sends authentication request with a valid Azure token            | Return 401 Unauthorized error and `RoleNotFound` message appears in log                  | Integration               | - [ ]            |
+| 4  | Conjur identity exists for Azure VM but doesn't have permissions on secret                   | VM sends request with valid Azure token to fetch secret             | Return 401 Unauthorized error and `RoleNotAuthorizedOnWebservice` message appears in log | Integration               | - [ ]            |
+| 1  | authn-azure is not whitelisted in ENV or enabled in DB                                       | VM sends authentication request to Conjur with valid Azure token    | Return 401 Unauthorized error and `AuthenticatorNotWhitelisted` message appears in log   | Integration               | - [ ]            |
+| 2  | Policy is loaded w/o `provider-uri` variable                                                 | VM sends authentication request to Conjur with valid Azure token    | Return 401 Unauthorized error and `RequiredResourceMissing` message appears in log       | Integration               | - [ ]            |
+| 2  | Policy is loaded w/o `provider-uri` value                                                    | VM sends authentication request to Conjur with valid Azure token    | Return 401 Unauthorized error and `RequiredSecretMissing` message appears in log         | Integration               | - [ ]            |
```

Comment:
nit: The policy is never loaded with a value. You can change to `Policy is loaded with `provider-uri` variable but a value is never loaded to it`. You can play with the wording.